crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -108,28 +108,62 @@ pub(crate) enum EnumVariantFields<'a> {
     Unit,
 }
 
+#[derive(Clone, Copy, PartialEq, Eq)]
+enum ReflectType {
```

Comment:
Nit/Bikeshed: Considering that reflection is all about the type system, I wonder if we should call this something like `ReflectMode` to be less ambiguous as to what "type" means.